### ISSUE TITLE
Your X-Clip or OpenAI vanilla CLIP?

### ISSUE THREAD

**Issue by Mut1nyJD** [lucidrains/DALLE2-pytorch#3]
First amazing that you try to recreate this. 
My question is are you going to plan to use your X-CLIP implementation or just use the basic OpenAI vanilla CLIP. From what I gathered from the official presentation I think most of the power of the DALLE-2l seems to lie in the clip embedding and obviously the move away from autoregression to diffusion and therefore also the prior diffusion into the CLIP embeddings. But OpenAI only gives out their rather weak clip models so its capabilities will be limited.

**Comment by lucidrains** [comment]
@Mut1nyJD sure! i can bring in x-clip :) https://github.com/lucidrains/x-clip (for those who do not know that line of research)

**Comment by lucidrains** [comment]
@Mut1nyJD i'm building out the prior network this morning :) should take 4-5 hours at most

**Comment by lucidrains** [comment]
loading up on :coffee: :coffee: :coffee: :coffee: :coffee:

**Comment by lucidrains** [comment]
@Mut1nyJD also, the group over at Laion have already replicated some of the CLIP models, and they will probably be on track on training a bigger better one. they are using openclip afaik. i'll probably aim to make the library compatible with all versions of CLIP, through some adapter or bridge class

**Comment by Mut1nyJD** [comment]
> @Mut1nyJD i'm building out the prior network this morning :) (i'm going for the best variant, the diffusion model that denoises the image embedding) should take 4-5 hours at most

Yeah this bit I find most confusing in terms of how they do it. What I don't quiet understood is this part DDIM and the Decoder is DDPM or are both DDIM.

**Comment by lucidrains** [comment]
spent all morning wrestling with jax jit problems 😐, i'll try to bring my mind to focus on it this afternoon!

**Comment by lucidrains** [comment]
@Mut1nyJD haha, i don't even know what DDIM is, i haven't been following the denoising literature that closely (but should probably start doing so)

**Comment by lucidrains** [comment]
also wow, so many github stars and i didn't even write any code yet lol

**Comment by MegaR** [comment]
> also wow, so many github stars and i didn't even write any code yet lol

For some reason it got in my google news feed lol. We will watch your career with great interest.

**Comment by lucidrains** [comment]
@MegaR how was it in google news? you probably mean OpenAI, who i'm not affiliated with

**Comment by MegaR** [comment]
@lucidrains No your repo was directly linked in the google assistant news feed for me. Like when you swipe left from the homescreen on pixel phones. It probably detected I was interested in DALL-E2

**Comment by lucidrains** [comment]
@MegaR oh wow, no way... i need to give google assistant news a try, it must have gotten a lot better

**Comment by lucidrains** [comment]
@MegaR oh wow, no way... google assistant news feed recommending a day-old repository without code is not something i would have guessed

**Comment by lucidrains** [comment]
alright, wrapped up https://github.com/lucidrains/palm-jax , going to knock out the prior network (the diffusion variant) by 6-7p, then go enjoy some san francisco nightlife

**Comment by lucidrains** [comment]
![20220408_144912](https://user-images.githubusercontent.com/108653/162536717-c3528bfc-30f5-475c-8a08-c88c3123c977.jpg)

Ice Cream agrees with that plan :)

**Comment by rom1504** [comment]
Supporting https://github.com/crowsonkb/cloob-training could be nice as well as it removes the need to do a text to image mapping, cloob space is already text/image aligned.
Also there will be some bigger cloobs.

**Comment by lucidrains** [comment]
haha, so many people emailing me about this repository... just relax, is there any reason to think i can't / won't do it? ;)

**Comment by lucidrains** [comment]
@rom1504 hi Romain, you follow me everywhere, even after joining Google lol

yea, x-clip should support CLOOB-like loss already

**Comment by lucidrains** [comment]
oh, do you mean to say CLOOB like architecture removed the need for the prior neural network? Is this just speculation atm?

**Comment by lucidrains** [comment]
ok, i think i'll save this for next week. measure twice cut once kind of thing. plus i probably shouldn't make it look too easy ;)

**Comment by lucidrains** [comment]
closing since i've already answered the original question

**Comment by rom1504** [comment]
> It's probably best to stick with the paper for now, and if the open source community can rid of the prior network with some modifications to CLIP, we need to verify with experimental comparisons between the two methods

Alstroemeria trained a cloob that is almost as good as clip and released it at the link I posted, the alignements between image and text embedding is better with it.
Is it enough to remove the needed of the prior network inside dalle 2 ? That's the hope but indeed it's just an hypothesis currently

Definitely agree that comparing the 2 methods will be interesting!

**Comment by lucidrains** [comment]
Yeah agreed! Katherine is doing great work as usual :)

**Comment by Mut1nyJD** [comment]
> @Mut1nyJD haha, i don't even know what DDIM is, i haven't been following the denoising literature that closely (but should probably start doing so)

As far as I know it is a different / more efficient way of sampling in terms of required time steps. But training scheme is the same between the two.

**Comment by lucidrains** [comment]
@Mut1nyJD ahh ok, let's forget about fast for now. there are so many papers trying to make DDPMs faster, and the dust hasn't totally settled yet
